Lib/test/test_hashlib.py

```diff
@@ -4,6 +4,7 @@
 #  Licensed to PSF under a Contributor Agreement.
 #
 
+import logging
```

Comment:
Can you order the import (namely put it after `import itertools`)? Thanks in advance.

Reply:
Can we add it as a lint rule for just this file?

Reply:
Wait, we have a `ruff` config for tests? I think the other import statements are also sorted so I think my comment applies to other files.

Reply:
Yes, there's only one rule at the moment, they have to be added one at a time

Reply:
the other files are not import sorted, and ruff finds 1058 issues across all the tests, I'll open a discussion topic to enable the import sort rule. I'm not prepared to manually import sort any files.

Reply:
I don't think this has anything to do with ruff. In general, we try to stay consistent with the current style of the file. I think we should follow that convention. In this specific file, the convention is to have sorted imports. It is also more convenient for future backports.

Reply:
yes I moved this import logging after import itertools, but I'm unable to tell if something is import sorted or not - I have to manually count it out on my fingers

Reply:
Oh, I see. I can review the few files now and tell you which ones need to be re-ordered, if there are any. As for the `ruff` rule itself, I would be in favor of adding that but I fear that, while accepted by the community, it would be rejected on the maintainer's side because it would create lots of conflicts during backports =/ (and we likely need to do that for all branches, including security branches, just to ease future backports). So it's likely to be a bit painful :(

Reply:
the great thing about import sort conflicts is they're easy to fix by accepting both changes and applying more import sorting

Reply:
possibly worth combining with [unused-import (F401)](https://docs.astral.sh/ruff/rules/unused-import/#unused-import-f401)

Reply:
> the great thing about import sort conflicts is they're easy to fix by accepting both changes and applying more import sorting

Yes but it's a burden that we need to implement on the miss-ilington's side. I would still be in favor of not breaking the current conventions (also, only 10 files were modified, so I think it's a small changeset compared to the work needed to be done afterwards)

Reply:
This would only be a cost once, after we backport the 1058 change PR to all the branches

Reply:
What about existing PRs? we don't really like introducing a conflict on existing PRs =/ (again, I'm not against doing it and if I could, I would just ruff the entire stdlib but I think we prefer the status quo than introducing maintainence burden). However, I'd be interested in hearing others thoughts.

Reply:
ah for existing PRs it would be worth enabling pre-commit.ci, then you click the "accept both changes" button in the github conflicts view and pre-commit.ci will resort and de-duplicate any imports via ruff.

Reply:
for a total format the world it's a bit trickier, but import sorting conflicts are much easier to solve automatically. I'm not sure how we'd get that through. Probably by starting with the tests and then moving through the stdlib module by module?

Reply:
Oh. So that could be an argument in favor that you can throw in the DPO thread because that's what is usually blocking such changes. Also, I know that some people prefer a certain import style so it could also be a per-file basis ruff rules for that. I think we can wrap up this discussion here and let the follow-up happen on the DPO thread.